### ISSUE TITLE
C-compatible QCSchema implementation

### ISSUE THREAD

**Issue by awvwgk** [MolSSI/QC_JSON_Schema#82]
I'm planning to create a QCSchema implementation which can be used easier with compiled languages. But before I set out to reinvent the wheel, does anyone know about an existing C-compatible QCSchema implementation?

**Comment by berquist** [comment]
No, but you might want to talk to @susilehtola: http://ccl.net/cgi-bin/ccl/message-new?2021+12+29+001

**Comment by susilehtola** [comment]
@berquist thanks for the tag.

Indeed, the more I've thought about this, the more I've realized that there is no perfect solution and we should just try to capitalize on QCSchema: there's no reason to invent many wheels since that will just add to the problem.

I am not very familiar with QCSchema, but I know that it can already store e.g. MO coefficients, meaning that the issue with various basis set orderings and normalizations already exists. @bennybp is there a standard for these data or is it just whatever the programs feed in?

If there is no standard, I fail to see the usefulness since then you can't pass the orbitals from one program to another...

**Comment by awvwgk** [comment]
I personally find QCSchema also somewhat lacking in features (periodic boundary conditions tops my list).

However it is at least one (open) standard that we have and that we could develop as community. But I fear without adaption outside of the QCEngine ecosystem there won't be no development.

I'm happy to push my software to support QCSchema, to show that it is possible and maybe also useful.

**Comment by ghutchis** [comment]
The point was to have at a minimum "a better Molden exchange." That was the baseline. I think we called it "bronze" level.

Frankly, the standard should specify basis set orderings and normalizations.

I don't think there are many independent implementations, besides `cclib`. I have some partial support in `Avogadro` but it's not in a released version.

**Comment by susilehtola** [comment]
> Frankly, the standard should specify basis set orderings and normalizations.

Yup, but this can be done in an "easier" way i.e. the necessary manipulations could be implemented in the QCSchema library so that the programs themselves don't need to be changed; instead, it would suffice to flag the used format in the QCSchema plugin and the interface could do the necessary juggling and normalization.

**Comment by susilehtola** [comment]
See also #45 on the ordering issue

**Comment by awvwgk** [comment]
I created an implementation of QCSchema Molecule in Fortran (https://github.com/grimme-lab/mctc-lib/) for xtb, dftd4, gcp, ... and other programs relying on the same IO library. Still, this is only a solution for one particular ecosystem of QC packages and I'm still interested in having a common library to share between packages.
